tests/ui/typeck/issue-81943.stderr

```diff
@@ -2,36 +2,32 @@ error[E0308]: mismatched types
   --> $DIR/issue-81943.rs:7:9
    |
 LL |   f(|x| lib::d!(x));
-   |         ^^^^^^^^^^ expected `()`, found `i32`
```

Comment:
(preexisting?) All three “*try adding a return type*” suggestions are syntactically malformed. Ideally, we would suggest (`-> $T { `, ` }`).

Reply:
Ah, sorry, I missed your last commit.